tests/ShadowDOM/html/activeElement.html

```diff
@@ -0,0 +1,344 @@
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_0);
+    assert.equal(r_0.shadowRoot.activeElement, r_0_0);
+    assert.equal(r_1.shadowRoot.activeElement, null);
+  });
+
+  test('r_0_0_l.focus()', function() {
+    r_0_0_l.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_0);
+    assert.equal(r_0.shadowRoot.activeElement, r_0_0_l);
+    assert.equal(r_1.shadowRoot.activeElement, null);
+  });
+
+  test('r_0_1.focus()', function() {
+    r_0_1.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_0);
+    assert.equal(r_0.shadowRoot.activeElement, r_0_1);
+    assert.equal(r_1.shadowRoot.activeElement, null);
+  });
+
+  test('r_0_1_l.focus()', function() {
+    r_0_1_l.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_0);
+    assert.equal(r_0.shadowRoot.activeElement, r_0_1_l);
+    assert.equal(r_1.shadowRoot.activeElement, null);
+  });
+
+  test('r_1.focus()', function() {
+    r_1.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1.shadowRoot.activeElement, null);
+  });
+
+  test('r_1_l.focus()', function() {
+    r_1_l.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1_l);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1_l.shadowRoot.activeElement, null);
+  });
+
+  test('r_1_0.focus()', function() {
+    r_1_0.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1.shadowRoot.activeElement, r_1_0);
+  });
+
+  test('r_1_0_l.focus()', function() {
+    r_1_0_l.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1.shadowRoot.activeElement, r_1_0_l);
+  });
+
+  test('r_1_1.focus()', function() {
+    r_1_1.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1.shadowRoot.activeElement, r_1_1);
+  });
+
+  test('r_1_1_l.focus()', function() {
+    r_1_1_l.focus();
+
+    assert.equal(doc.activeElement, r);
+    assert.equal(r.shadowRoot.activeElement, r_1);
+    assert.equal(r_0.shadowRoot.activeElement, null);
+    assert.equal(r_1.shadowRoot.activeElement, r_1_1_l);
+  });
+
+  test('focused element in different document', function() {
+    var docType = doc.implementation.createDocumentType(
+      'html', '-//W3C//DTD XHTML 1.0 Transitional//EN',
+      'http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd');
+    var otherDocument = doc.implementation.createDocument(
```

Comment:
Actually, this whole test shouldn't matter. `document.activeElement` shouldn't change if you manually focus an element in a different document.